client/src/controllers/PreviewController.ts

```diff
@@ -0,0 +1,789 @@
+import axe from 'axe-core';
+
+import { Controller } from '@hotwired/stimulus';
+import type { ContextObject } from 'axe-core';
+import {
+  getAxeConfiguration,
+  getA11yReport,
+  renderA11yResults,
+} from '../includes/a11y-result';
+import { wagtailPreviewPlugin } from '../includes/previewPlugin';
+import {
+  getPreviewContentMetrics,
+  renderContentMetrics,
+} from '../includes/contentMetrics';
+import { WAGTAIL_CONFIG } from '../config/wagtailConfig';
+import { debounce } from '../utils/debounce';
+import { gettext } from '../utils/gettext';
+import type { ProgressController } from './ProgressController';
+import { setOptionalInterval } from '../utils/interval';
+
+const runContentChecks = async () => {
+  axe.registerPlugin(wagtailPreviewPlugin);
+
+  const contentMetrics = await getPreviewContentMetrics({
+    targetElement: 'main, [role="main"], body',
+  });
+
+  // This requires Wagtail's preview plugin for axe to be registered in the
+  // preview iframe, which is not done in tests as the registration happens via
+  // the userbar.
+  if (!contentMetrics) return;
+
+  renderContentMetrics({
+    wordCount: contentMetrics.wordCount,
+    readingTime: contentMetrics.readingTime,
+  });
+};
+
+const runAccessibilityChecks = async (
+  onClickSelector: (selectorName: string, event: MouseEvent) => void,
+) => {
+  const a11yRowTemplate = document.querySelector<HTMLTemplateElement>(
+    '#w-a11y-result-row-template',
+  );
+  const checksPanel = document.querySelector<HTMLElement>(
+    '[data-checks-panel]',
+  );
+  const config = getAxeConfiguration(document.body);
+  const toggleCounter = document.querySelector<HTMLElement>(
+    '[data-side-panel-toggle="checks"] [data-side-panel-toggle-counter]',
+  );
+  const panelCounter = document.querySelector<HTMLElement>(
+    '[data-side-panel="checks"] [data-a11y-result-count]',
+  );
+
+  if (
+    !checksPanel ||
+    !a11yRowTemplate ||
+    !config ||
+    !toggleCounter ||
+    !panelCounter
+  ) {
+    return;
+  }
```

Comment:
This previously did not check for `checksPanel` which would crash my tests because I don't have it in the test markup.